BUILD.md

```diff
@@ -0,0 +1,86 @@
+# Important notice
+***In order to uniformize and simplify the build system we had to make choices. TC is currently only officially supported on Ubuntu 16.04 with gcc 5.4.0, cuda 9.0 and cudnn 7.***
+Other configurations may work too but are not yet officially supported.
+For more information about setting up the config that we use to build the conda dependencies see the following [Dockerfile](conda_recipes/Dockerfile).
+
+Our main goal with this decision is to make the build procedure extremely simple, reproducible both internally and extensible to new targets in the future.
+In particular the gcc-4 / gcc-5 ABI switch is not something we want to concern ourselves with at this point, we go for gcc-5.4.0.
+
+# Conda from scratch (one time config)
+Choose and set a CONDA_PATH then run the following:
+
+```
+wget https://repo.anaconda.com/archive/Anaconda3-5.1.0-Linux-x86_64.sh -O anaconda.sh && \
+    chmod +x anaconda.sh && \
+    ./anaconda.sh -b -p ${CONDA_PATH} && \
+    rm anaconda.sh
+
+. ${CONDA_PATH}/bin/activate
+conda update -y -n base conda
+```
+
+Create a new environment in which TC will be built and install core dependencies:
+```
+conda create -y --name tc_build python=3.6
+conda activate tc_build
+conda install -y pyyaml mkl-include pytest
+conda install -y -c pytorch pytorch torchvision cuda90
+conda remove -y cudatoolkit --force
+conda install -y -c nicolasvasilache llvm-tapir50 halide
+```
+
+# Optional dependencies
+Optionally if you want to use Caffe2:
+```
+conda install -y -c conda-forge eigen
+conda install -y -c nicolasvasilache caffe2
+```
+
+# Activate preinstalled conda in your current terminal
+```
+. ${CONDA_PATH}/bin/activate
+conda activate tc_build
+```
+
+# Cudnn version
+The cudnn version is currently cast in stone by the caffe2 dependency.
+This comes directly from the version that NVIDIA imposes on us via their supported
+[Dockerfile](conda_recipes/docker-images/tc-cuda9.0-cudnn7.1-ubuntu16.04-devel/Dockerfile).
+
+For now this is cudnn 7.1.
+If one cannot install cudnn 7.1 system-wide, one may resort to the following:
+```
+conda install -c anaconda cudnn
+conda remove -y cudatoolkit --force
+```
+
+# Build TC with dependencies supplied by conda (including cudnn 7.1)
+```
+CLANG_PREFIX=$(${CONDA_PREFIX}/bin/llvm-config --prefix) ./build.sh
+```
+Alternatively you may need to pass the environment variable `CUDA_TOOLKIT_ROOT_DIR` pointing
```

Comment:
"Additionally", rather than "alternatively".  Build script needs `CLANG_PREFIX` independently of `CUDA_TOOLKIT_ROOT_DIR`.  Specifying CUDA path is also related to my question above about removing cudatoolkit.